api_tests/base/test_sloan_study.py

```diff
@@ -0,0 +1,86 @@
+import pytest
+import mock
+
+from waffle.models import Flag
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    PreprintFactory
+)
+
+from osf.features import (
+    SLOAN_DATA,
+    SLOAN_PREREG,
+    SLOAN_COI
+)
+
+SLOAN_FLAGS = (
+    SLOAN_DATA,
+    SLOAN_PREREG,
+    SLOAN_COI
+)
+
+@pytest.mark.django_db
+class TestSloanStudyWaffling:
```

Comment:
Could you do a test where an unauthenticated user has all false cookies, then logs in with a user that has all true flags, to verify that the cookies are then all true?